### ISSUE TITLE
Dealing with "big" update

### ISSUE THREAD

**Issue by psmiraglia** [saltstack-formulas/logstash-formula#25]
In the current status, the formula results to be a bit outdated (e.g. old DEB repository) and uses a monolithic approach (a single init state that installs and configures everything).

Recently I had to install the latest version of logstash (5.4.x) and this formula resulted to be not usable, so I had to rewrite it in a "modern" way. Here the repo

https://github.com/psmiraglia/logstash-formula

In this case, where a massive update could be usefull for the community, how to deal with PR? I mean, is considered licit a PR that totally rewrite the formula?

**Comment by johnkeates** [comment]
I too have a much better formula for this, it's not had any maintenance for a year or so, but we should indeed update this formula.

https://github.com/johnkeates/logstash-formula

I think we can do a total rewrite as long as calling init still does "everything". Configuration-wise, we should probably not destroy the current pillar format. We can add stuff, so we could use a key for non-logstash configurations (i.e. sysconfig and defaults) so that it doesn't interfere with the logstash config itself.

Maybe I can PR yours or you can PR mine and then we PR the result here.

**Comment by psmiraglia** [comment]
It sounds good...

Since your formula is an enhanced fork of the original one (while mine was defined from the scratch), I think PRing you with my stuff is the "natural" flow.

How to deal with input/filter/output configurations? Should we maintain the original approach (pillar data translated in LS directive/syntax)? Is it really needed? Is it really a sustainable solution?

**Comment by psmiraglia** [comment]
Here the PR with my features (https://github.com/johnkeates/logstash-formula/pull/1)

**Comment by psmiraglia** [comment]
Hey @johnkeates, any news about the PR?

**Comment by johnkeates** [comment]
Yeah, sorry about that, lots of GitHub notifications and so little time... I'll sort it out today/tomorrow!

**Comment by absmith82** [comment]
also addressed with PR#29
